internal/controller/apiextensions/composite/composed.go

```diff
@@ -159,47 +160,83 @@ func NewAPIConnectionDetailsFetcher(c client.Client) *APIConnectionDetailsFetche
 }
 
 // FetchConnectionDetails of the supplied composed resource, if any.
-func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) {
+func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) { // nolint:gocyclo
 	sref := cd.GetWriteConnectionSecretToReference()
+	var data map[string][]byte
 	if sref == nil {
-		return nil, nil
+		// We hit this block if the composed resource does not specify a secret ref
+		// this would occur if it does not publish any secrets, however we may want to still
+		// publish some values or access values from fieldPaths.
+		data = make(map[string][]byte)
+	} else {
+		// It's possible that the composed resource does want to write a
+		// connection secret but has not yet. We presume this isn't an issue and
+		// that we'll propagate any connection details during a future
+		// iteration.
+		s := &corev1.Secret{}
+		nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
+		if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
+			return nil, errors.Wrap(err, errGetSecret)
+		}
+		data = s.Data
 	}
-
 	conn := managed.ConnectionDetails{}
 
-	// It's possible that the composed resource does want to write a
-	// connection secret but has not yet. We presume this isn't an issue and
-	// that we'll propagate any connection details during a future
-	// iteration.
-	s := &corev1.Secret{}
-	nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
-	if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
-		return nil, errors.Wrap(err, errGetSecret)
-	}
-
 	for _, d := range t.ConnectionDetails {
-		if d.Name != nil && d.Value != nil {
-			conn[*d.Name] = []byte(*d.Value)
-			continue
+		switch d.Type {
+		case v1.ConnectionDetailValue:
+			if d.Name != nil && d.Value != nil {
+				// Name must be set if value type
+				conn[*d.Name] = []byte(*d.Value)
+			}
```

Comment:
I wonder whether it would make sense for this case to return an error if name or value are missing. We now have a clear understanding of the author's intent - they're declaring that this is a `type: Value` connection secret, which requires a name and a value. It might be useful to return an error if one or both are missing rather than silently proceeding without the requested connection detail.

Note that this case is different from the following two because in those cases the connection secret key or composed resource field value are likely to be absent at first, before eventually being populated. In those cases we can't tell the difference between "the value hasn't appeared yet" and "the value will never appear", but in this one we know that the value will never show up without human intervention.

Reply:
Hmm that makes a lot of sense, I'll add in the corresponding error + test now

Reply:
Similarly, should I return an error if the name is not set for `v1.ConnectionDetailFromFieldPath`, since this also expects the name to be set?

Reply:
Done, let me know what you think :)